api/src/main/java/org/pmiops/workbench/exfiltration/EgressRemediationService.java

```diff
@@ -1,6 +1,5 @@
 package org.pmiops.workbench.exfiltration;
```

Comment:
This class is refactored to provide two implementations, one for sumologic, and the other for the file lengths. 
The main change is to change the class into abstract, and change it to a template pattern. The two abstract methods are implemented differently in the specific classes